NYPL-discovery/discovery-front-end
# Scc 2536

**What's this do?**
- Adds a warning modal when user is going to be timed out
- Changes the trigger for redirecting the patron from a fixed timeout to the iframe's `onload` event.

A slight gotcha: there are various things you can do that will trigger a timer reset, like adjusting the size of the screen. This happens because various things trigger a React re-render, which in turn resets the timer. I actually think this is correct behavior, since we want to keep the user logged in if they are still using the page, but it means anyone testing this has to actually do nothing on the page for a clean test.

**Why are we doing this? (w/ JIRA link if applicable)**
This is scc-2536

**Do these changes have automated tests?**

Yes. The tests are not ideal because they 1. involve a lot of complicated set up/tear down for mocks 2. use a style of mocking that should probably be replaced by more sinon's built in mock functionality and 3. are somewhat rigid. 

That said, this is a complicated component to test. I am open to suggestions for improving the tests, but I also don't think we should let improving the test suite hold up getting this component in front of QA for too long.

**How should this be QAed?**
- Navigate to accounts page
- after 2 minutes, you should see the modal
- if you click to stay logged in, timer should reset and modal should go away
- if you click to log out, you should be redirected to the landing page. If you go back to accounts, you should be prompted to log in again
- if you do nothing, you should be redirected to the landing page. If you go back to accounts, you should be prompted to log in again

If you are running this locally. you can speed up the QA by changing the logout time to something smaller.

**Dependencies for merging? Releasing to production?**

**Has the application documentation been updated for these changes?**

**Did someone actually run this code to verify it works?**

PR author tested locally.
